solver/src/driver.rs

```diff
@@ -258,6 +258,27 @@ impl Driver {
         }
     }
 
+    // Log the amount of orders that were matched but not settled in this runloop (effectively queued for the next one)
+    // Should help us to identify how much we can save by parallelizing execution.
+    fn report_matched_but_unsettled_orders<I: Iterator<Item = Settlement>>(
+        &self,
+        submitted: &Settlement,
+        all: I,
+    ) {
```

Comment:
```suggestion
    fn report_matched_but_unsettled_orders(
        &self,
        submitted: &Settlement,
        all: impl Iterator<Item = Settlement>,
    ) {
```

Reply:
https://doc.rust-lang.org/stable/book/ch10-02-traits.html#traits-as-parameters

It means the same thing but is easier to write.